### PR TITLE
Ensure focus is on cancel button for Reset Form

### LINKED ISSUE #8467: Set "Cancel" button as the Reset prompt default
## Feature description

So, I use "R" a lot to open the "Reset Changes" prompt and resetting some of the files that I didn't want to change, but I've ran into a problem today, right on the Commits window, where the button labeled as "Reset" on the "Reset Changes" prompt is focused instead of the button labeled as "Cancel" 

Pressing "R" pops up this prompt asking you if you want to reset your changes, but double tapping "R" accepts which ever is highlighted, creating a mistake-prone behaviour

So, I pressed "R" but my finger slipped away a second time, hitting "R" again, successfully aproving the prompt and consequely, resetting my changes.

And so, I reset all the changes from my files, without even noticing

I wish that this behaviour is set to the opposite action, cancel, rather than a quick way of resetting changes cause, it's easier to click it again and say "I want to reset my changes" rather than "Now I have to redo everything because of my mistake"

Thank you, very much!

Screenshot_1

## Environment

- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.19041.0
- .NET Framework 4.8.4220.0
- DPI 96dpi (no scaling)

### PR DESCRIPTION
**Pull request by BrianFreemanAtlanta**
Fixes #8467 

## Proposed changes
- Added `btnCancel.Select()` to the constructor. Left existing mnemonic of "r" for the reset button.
- "Focus" did not stick after form was shown, but after testing found that "select" worked during form load/initialization

### Before

image

### After
image

## Test methodology 
- I currently performed manual UI tests both visual and then message feedback (removed after test) to verify focus. I can add UI test if desired.

## Test environment(s) 

- GIT 2.28
- Windows 10

As mentioned in the issue report - user reported they pushed r twice and files reset. This will not change that behavior due to mnemonic, but will prevent "enter" key from automatically resetting files.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by pmiossec** [comment]
Is the second screenshot wrong? Because I don't see a difference...

**Comment by BrianFreemanAtlanta** [comment]
I edited and updated the after screenshot. The "Focus" didn't work had to revert to "Select"

**Review by gerhardol — COMMENTED** [review]
ok

File: GitUI/CommandsDialogs/FormResetChanges.cs [hunk 1]
```diff
@@ -40,6 +40,8 @@ public FormResetChanges(bool hasExistingFiles, bool hasNewFiles)
             {
                 cbDeleteNewFilesAndDirectories.Enabled = true; // A mix of types, so enable the checkbox.
             }
+
+            btnCancel.Select(); // Focus doesn't stick while form is loading/not shown, select does.
```
Comment (pmiossec) [inline GitUI/CommandsDialogs/FormResetChanges.cs:44]: When I see this comment, I wonder if we'd add the `Focus()` call in the 'Shown' event

Comment (BrianFreemanAtlanta) [reply GitUI/CommandsDialogs/FormResetChanges.cs:44]: I will check it out in the morning see if.that will work.

Comment (RussKie) [reply GitUI/CommandsDialogs/FormResetChanges.cs:44]: We have `RuntomeLoaded` (or something like that) that is used to init state when a dialog opens.

Comment (RussKie) [reply GitUI/CommandsDialogs/FormResetChanges.cs:44]: Also the whole is probably fixable by changing the tabindex of buttons in the designer file.

Comment (oriash93) [reply GitUI/CommandsDialogs/FormResetChanges.cs:44]: > Also the whole thing is probably fixable by changing the tabindex of buttons in the designer file.

Definitely

Comment (BrianFreemanAtlanta) [reply GitUI/CommandsDialogs/FormResetChanges.cs:44]: It was the long way around, but I changed the tabindex & that does let the focus be on cancel when the form appears.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Please rebase resolving the conflict, and squash. And it is good to go.
Thank you

**Comment by pmiossec** [comment]
@BrianFreemanAtlanta Please avoid solving the conflicts by merging with master. We prefer, as @RussKie asked,that you rebase and squash to keep only one commit and makes the history much more readable.

**Comment by BrianFreemanAtlanta** [comment]
That was a learning attempt to resolve the conflict in contributors.txt. I am working at attempting to rebase and squash my checkins. It's a learning experience.

**Comment by gerhardol** [comment]
> That was a learning attempt to resolve the conflict in contributors.txt. I am working at attempting to rebase and squash my checkins. It's a learning experience.

Still a problem..
Merges are evil.

Reset soft on master, commit and push force-with-lease

**Comment by gerhardol** [comment]
Still a merge commit...
The commit history is hard to follow if the branches to merge includes merges

**Comment by RussKie** [comment]
> 2 commits - update my fork & actual work.

Could you please clarify what value has 3083983 to this PR or the project in general?

I'd like to see a single commit with your changes that relate to fixing #8467. Nothing else.

**Comment by RussKie** [comment]
I added a wiki: https://github.com/gitextensions/gitextensions/wiki/How-To:-Squash-and-Rebase-your-changes

**Comment by RussKie** [comment]
Please completely revert changes to contributors.txt in your changeset, and it is good to go.

**Review by mstv — APPROVED** [review]

**Comment by gerhardol** [comment]
@msftbot merge in 1 day

**Comment by RussKie** [comment]
Thank you!
